### ISSUE TITLE
Problems with LXD on Fedora 26

### ISSUE THREAD

**Issue by iaroki** [ganto/copr-lxd#8]
Hi, instructions on your copr page works perfectly with Fedora 25 but on Fedora 26 after all steps `lxd.service` doesn't start. The problem with `lxd` daemon in:

```
[hostmorke@stormey ~]$ systemctl status lxd
● lxd.service - LXD - main daemon
   Loaded: loaded (/usr/lib/systemd/system/lxd.service; indirect; vendor preset: disabled)
   Active: activating (start-post) (Result: exit-code) since Mon 2017-10-02 14:57:47 EEST; 1min 15s ago
     Docs: man:lxd(1)
  Process: 2750 ExecStart=/usr/bin/lxd --group lxd --logfile=/var/log/lxd/lxd.log (code=exited, status=1/FAILURE)
 Main PID: 2750 (code=exited, status=1/FAILURE); Control PID: 2751 (lxd)
    Tasks: 5
   Memory: 18.8M
      CPU: 119ms
   CGroup: /system.slice/lxd.service
           └─control
             └─2751 /usr/bin/lxd waitready --timeout=600

Oct 02 14:57:47 stormey systemd[1]: Starting LXD - main daemon...
Oct 02 14:57:49 stormey lxd[2750]: lvl=warn msg="AppArmor support has been disabled because of lack of kernel support" 
Oct 02 14:57:50 stormey lxd[2750]: error: failed to fetch update versions: query yields  column, not INTEGER
Oct 02 14:57:50 stormey systemd[1]: lxd.service: Main process exited, code=exited, status=1/FAILURE
```

Manual running `lxd` binary gives this error:

```
[hostmorke@stormey ~]$ sudo /usr/bin/lxd --group lxd --logfile=/var/log/lxd/lxd.log
WARN[10-02|15:02:51] AppArmor support has been disabled because of lack of kernel support 
error: failed to fetch update versions: query yields  column, not INTEGER
```

Any advice?

**Comment by ganto** [comment]
I guess you are using the latest `lxd-2.18-2.fc26`? Is this a fresh installation?

Can you output the result of:
```
sqlite3 /var/lib/lxd/lxd.db 'SELECT version FROM schema ORDER BY version;'
```
(that's the actual query that is failing)

On my system (Fedora 25), I get the following output:
```
34
35
36
```

Can you backup `/var/lib/lxd/lxd.db`, delete it and start `lxd` again to see if this can be reproduced?

**Comment by iaroki** [comment]
Yes, latest packages from your copr repo and two different fresh installs of Fedora 26 gives that error.
I have working Fedora 25 and `lxd` work perfectly. The problem is on Fedora 26.

```
[root@stormey ~]# sqlite3 /var/lib/lxd/lxd.db 'SELECT version FROM schema ORDER BY version;'
36
```

After deleting `/var/lib/lxd/lxd.db` the error still there:

```
[root@stormey ~]# 
[root@stormey ~]# sqlite3 /var/lib/lxd/lxd.db 'SELECT version FROM schema ORDER BY version;'
36
[root@stormey ~]# mv /var/lib/lxd/lxd.db /var/lib/lxd/lxd.db.bak
[root@stormey ~]# /usr/bin/lxd --group lxd --logfile=/var/log/lxd/lxd.log
WARN[10-03|13:12:35] AppArmor support has been disabled because of lack of kernel support 
error: failed to fetch update versions: query yields  column, not INTEGER
[root@stormey ~]# sqlite3 /var/lib/lxd/lxd.db 'SELECT version FROM schema ORDER BY version;'
36
[root@stormey ~]#
```

Can you test it on Fedora 26. It is strange that it works on 25 and doesn't on 26.

```
[root@stormey ~]# rpm -q lxd lxd-client lxd-tools
lxd-2.18-2.fc26.x86_64
lxd-client-2.18-2.fc26.x86_64
lxd-tools-2.18-2.fc26.x86_64
[root@stormey ~]# uname -a
Linux stormey 4.12.14-300.fc26.x86_64 #1 SMP Wed Sep 20 16:28:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[root@stormey ~]# cat /etc/os-release 
NAME=Fedora
VERSION="26 (Workstation Edition)"
ID=fedora
VERSION_ID=26
PRETTY_NAME="Fedora 26 (Workstation Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:26"
HOME_URL="https://fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=26
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=26
PRIVACY_POLICY_URL=https://fedoraproject.org/wiki/Legal:PrivacyPolicy
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

**Comment by ganto** [comment]
Ok, I can also reproduce the issue on Fedora 26. Thanks a lot for the report. As you can think, I still use Fedora 25 for my setup so I didn't notice before.

However, I have no idea what is going wrong here. I opened an issue upstream, see lxc/lxd#3888

**Comment by iaroki** [comment]
I see you solved problem with `go-sqlite3`. Any chances to rebuild and update `lxd` packages in your repo soon?

**Comment by ganto** [comment]
The fixed version `lxd-2.18-3` should be available now. 😃

**Comment by iaroki** [comment]
Thanks a lot for feedback and solving the problem. Everything works perfectly now  :thumbsup:
